tests/suite_test.go

```diff
@@ -176,23 +195,83 @@ func (suite *IntegrationTestSuite) SetupSuite() {
 	require.NoError(suite.T(), err, fmt.Sprintf("Error Deploying DSPA:\n%s", testUtil.PrintConditions(ctx, DSPA, DSPANamespace, clientmgr.k8sClient)))
 	loggr.Info("DSPA Deployed.")
 
-	loggr.Info("Setting up Portforwarding service.")
-	options := []*forwarder.Option{
-		{
-			LocalPort:   PortforwardLocalPort,
-			RemotePort:  APIServerPort,
-			ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
-			Namespace:   DSPANamespace,
-		},
+	if endpointType == "service" {
+
+		loggr.Info("Setting up Portforwarding service.")
+		options := []*forwarder.Option{
+			{
+				LocalPort:   PortforwardLocalPort,
+				RemotePort:  APIServerPort,
+				ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
+				Namespace:   DSPANamespace,
+			},
+		}
+
+		forwarderResult, err = forwarder.WithForwarders(ctx, options, kubeconfig)
+		suite.Require().NoError(err)
+		_, err = forwarderResult.Ready()
+		suite.Require().NoError(err)
+
+		APIServerURL = fmt.Sprintf("http://127.0.0.1:%d", PortforwardLocalPort)
+
+		loggr.Info(fmt.Sprintf("Port forwarding service Successfully set up: %s", APIServerURL))
+
+	} else if endpointType == "route" {
+
+		// Define the namespaced name
+		key := types.NamespacedName{
+			Namespace: DSPANamespace,
+			Name:      "ds-pipeline-" + suite.DSPA.Name,
+		}
+
+		route := &routev1.Route{}
+
+		// Retrieve the router
+		count := 1
+		ok := false
+		for count < 10 {
+			err := clientmgr.k8sClient.Get(context.TODO(), key, route)
+			if err == nil {
+				ok = true
+				break
+			}
+			count++
+			time.Sleep(1 * time.Second)
+		}
+		if !ok {
+			log.Fatal("failed to retrieve router")
+		}
+
+		APIServerURL = fmt.Sprintf("https://%s", route.Status.Ingress[0].Host)
+
+		loggr.Info(fmt.Sprintf("The Api Server URL was retrieved from the Route: %s", APIServerURL))
+
+		suite.Clientmgr.httpClient.Transport = &customTransport{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+			Token: cfg.BearerToken,
+		}
+
+		// waiting for pods to sit down
+		count = 1
+		ok = false
+		for count < 10 {
+			response, _ := suite.Clientmgr.httpClient.Get(fmt.Sprintf("%s/apis/v2beta1/healthz", APIServerURL))
+			if response.StatusCode == 200 {
+				ok = true
+				break
+			}
+			count++
+			time.Sleep(1 * time.Second)
+		}
+		if !ok {
+			log.Fatal("Pods didn't started properly")
+		}
```

Comment:
this is already covered with this test here: https://github.com/opendatahub-io/data-science-pipelines-operator/pull/711/files#diff-d430921125862c61b7f677a4fa66220726faac5ef8827551fb3accd3ee5c1bd1R32


pod readiness check already queries the healthz endpoint

Reply:
What you wrote is the desirable state but it is not working like that. The code was implemented after seeing failures. We can address and fix it later. The more we really in the readiness check the better.

Reply:
okay, in that case my suggestion is to re-use the waitFor logic that already exists instead of re-implementing it here, see: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/54c54dd248a6c03ddac6760f4c86c7f0d9b40ddb/tests/util/resources.go#L126

Reply:
you could also just adjust the condition for what a successful deployDSPA is here https://github.com/opendatahub-io/data-science-pipelines-operator/pull/711/files#diff-96e66e1ccb0083ac5a2b7fc37cb002803d242ea67c1981a5d09ef8ee2668354cR189 

and consider the Ready condition as well